packages/@aws-cdk/aws-codeartifact/lib/validation.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Throws an error if a value is defined and not an integer or not in a range.
+ */
+export function validate(name: string, rule : { required?: boolean, pattern? : RegExp, minLength? : number, maxLength?: number }, value?: string) {
+  if (rule.required && value === undefined || /^s{1,}$/gi.test(value || '')) {
+    throw new Error(`${name}: Is required and cannot be empty`);
+  }
+
+  if (value === undefined) { return; }
+
+  if (!rule.required && (value === undefined || value === '')) { return; }
```

Comment:
Your regex is broken; it's missing a `\` before the `s`, but also only checks for strings with one or more spaces; an empty string will pass. I think the simpler way to handle these three checks is:

```suggestion
  if (!value) {
    if (rule.required) {
      throw new Error(`'${name}' is required and cannot be empty`);
    }
    return;
  }
```